runtime/src/main/java/io/quarkiverse/githubaction/runtime/github/GitHubRepositoryFactory.java

```diff
@@ -0,0 +1,30 @@
+package io.quarkiverse.githubaction.runtime.github;
+
+import java.io.IOException;
+
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GitHub;
+
+import io.quarkiverse.githubaction.Context;
+
+public record GitHubRepositoryFactory(GitHub gitHub, Context ctx) {
```

Comment:
I think I would avoid this level of indirection and have this code in `GitHubEvent`. I don't think it would make it a lot more complex - except if I missed something.

Reply:
Integrated in `GitHubEvent`